ktor-http/src/io/ktor/http/URLBuilder.kt

```diff
@@ -86,6 +84,10 @@ data class Url(
     val password: String?,
     val trailingQuery: Boolean
 ) {
+    init {
+        require(port >= 0 || port == UNKNOWN_PORT) { "port must be >= 0 or $UNKNOWN_PORT if not set" }
```

Comment:
Should `0` be allowed as a port? Perhaps this should actually validate a legitimate range
```kotlin
require(port in 1..65535 || port == UNKNOWN_PORT)
```